lib/fluent/log.rb

```diff
@@ -53,9 +53,17 @@ def self.str_to_level(log_level_str)
       end
     end
 
-    def initialize(out=STDERR, level=LEVEL_TRACE, opts={})
-      @out = out
-      @level = level
+    def initialize(logger, opts={})
+      # overwrites logger.level= so that config reloading resets level of Fluentd::Log
+      orig_logger_level_setter = logger.class.public_instance_method(:level=).bind(logger)
+      me = self
+      # DaemonLogger's log level setting has difference of one level from that of this class.
+      # Then level in this class will be self.level + 1.
```

Comment:
I had a doubt that:
* `level` of Fluentd logger is a number, between 0-5
* the `+1` means ServerEngine's logger level is also a number, but between 1-6.
* is it "really" collect?

Fluentd engineer might be unfamiliar to ServerEngine (because these are different product :)), so we should add comments about ServerEngine's log levels especially for the case like doing `+1` on Fluentd's log level.